lib/takeaway.rb

```diff
@@ -0,0 +1,32 @@
+class Takeaway
```

Comment:
This class `Takeaway` could probably benefit to be split into smaller classes (at least two) to respect what's called separation of concerns — right know this class is doing different things:
 * it holds the menu (with the instance variable `@menu`)
 * it can print the menu
 * it holds the order and can add something to it (`@orders` and method `add_orders`)
 * it can print the order content (and the total)

The easiest would be to start by extracting everything related to the menu (so the 2 first points) into a dedicated class (maybe called `Menu`), and this class would be responsible for storing the menu and printing it.

The `Takeaway` class would then become the "main" class of our program, and would hold an instance variable `@menu` that is an instance of the menu class — which means the initialize method would then look like this:

```ruby
def initialize
    @menu = Menu.new
    @orders = []
    @sum = 0
end
```

... then, when we need to access the menu (for example, in the method `print_orders`), we could use a method `menu_items` of the `Menu` class:

```ruby
def print_orders
  # @menu is not a hash anymore, it's an instance of the new Menu class
  @menu.menu_items.each do |key, value|
    # etc...
  end
end
```

If you're not sure where to start, try these steps:
  * create a new `Menu` class in another file
  * start copying the methods related to menu management into this class
  * remove them from the `Takeaway` class
  * make the necessary changes so `Takeaway` now *relies on the new `Menu` methods *